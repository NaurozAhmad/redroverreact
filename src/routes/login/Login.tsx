import styles from './Login.module.css';

import Navbar from 'components/navbar/Navbar';
import { Box, Button, TextField } from '@mui/material';

const Login = () => {
  return (
    <>
      <Navbar />
      <div className={styles['content']}>
        <h1>Login</h1>
        <Box component="form" sx={{ '& > :not(style)': { mb: 1, width: '100%' } }}>
          <TextField id="email" label="Email" type="email" variant="outlined" />
          <TextField id="password" label="Password" type="password" variant="outlined" />
        </Box>
        <div className="mt1">
          <Button variant="contained" fullWidth>
            Log in
          </Button>
        </div>
      </div>
    </>
  );
};

export default Login;
