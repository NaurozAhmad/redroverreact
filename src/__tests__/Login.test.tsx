import { render, screen, cleanup, act } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import Login from 'routes/login/Login';

let container: any = null;

test('should render login page', () => {
  act(() => {
    container = render(
      <BrowserRouter>
        <Login />
      </BrowserRouter>
    ).container;
  });
  // container = render(<Login />).container;
  // expect(container).toMatchSnapshot();
  expect(screen.getByText('Login')).toBeInTheDocument();
  expect(screen.getByTestId('loginform')).toBeInTheDocument();
  expect(screen.getByRole('textbox', { name: 'Email' })).toBeInTheDocument();
  expect(screen.getByTestId('l-password')).toBeInTheDocument();
  expect(screen.getByRole('button', { name: 'Log in' })).toBeInTheDocument();
});
