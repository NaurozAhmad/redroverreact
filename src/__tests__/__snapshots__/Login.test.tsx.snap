// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render login page 1`] = `
<div>
  <header
    class="navbar"
  >
    <span
      class="title"
    >
      Red Rover
    </span>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium button css-1e6y48t-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </header>
  <div
    class="content"
  >
    <h1>
      Login
    </h1>
    <form
      class="MuiBox-root css-2ff0th"
    >
      <div
        class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="false"
          for="email"
          id="email-label"
        >
          Email
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
            id="email"
            name="email"
            type="email"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-yjsfm1"
            >
              <span>
                Email
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <div
        class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="false"
          for="password"
          id="password-label"
        >
          Password
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
            id="password"
            name="password"
            type="password"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-yjsfm1"
            >
              <span>
                Password
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <div
        class="mt1"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth css-1fu7jd5-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="submit"
        >
          Log in
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </form>
  </div>
</div>
`;
